### PR TITLE
changes for tls 1.3 with FIPS ready

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
ZD16581

### CODE REVIEW

**Review by SparkiDev — APPROVED** [review]

File: src/wp_wolfprov.c [hunk 1]
```diff
@@ -997,6 +997,10 @@ int wolfssl_provider_init(const OSSL_CORE_HANDLE* handle,
     int ok = 1;
     OSSL_FUNC_core_get_libctx_fn* c_get_libctx = NULL;
 
+#ifdef HAVE_FIPS
+    PRIVATE_KEY_UNLOCK();
```
Comment (SparkiDev) [inline src/wp_wolfprov.c:1001]: There are unlocks and locks around every private key operation.
Why is this needed and is there a lock call?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/wp_wolfprov.c (hunk 1 above)
Comment (JacobBarthelmeh) [reply src/wp_wolfprov.c:1001]: It's needed for the HKDF extract call. It could be placed just around that function call.

Comment (SparkiDev) [reply src/wp_wolfprov.c:1001]: Please do.
I don't want global unlocks going on.
Thanks!

Comment (JacobBarthelmeh) [reply src/wp_wolfprov.c]: updated, thanks for the review

**Review by SparkiDev — APPROVED** [review]
